File: council/prompt.py
from pathlib import Path

PROMPTS_DIR = Path(__file__).resolve().parent.parent / "prompts"


def _load(name: str) -> str:
    return (PROMPTS_DIR / name).read_text().strip()


def discussion_prompt(round_num: int, total_rounds: int) -> str:
    """Generate the system context for a discussion round."""
    round_guidance = _load(f"round-{round_num}.md")
    engagement = _load("engagement.md")
    brevity = _load("brevity.md")
    response_format = _load("response-format.md")

    return (
        f"You are participating in a council discussion, Round {round_num} of {total_rounds}.\n\n"
        f"{round_guidance}\n\n"
        f"{engagement}\n\n"
        f"{brevity}\n\n"
        f"{response_format}"
    )


def vote_prompt(question: str) -> str:
    """Generate the system context for the vote round."""
    template = _load("vote.md")
    engagement = _load("engagement.md")
    vote_format = _load("vote-format.md")

    return (
        f"{template.replace('{question}', question)}\n\n"
        f"{engagement}\n\n"
        f"{vote_format}"
    )

File: src/prompt.rs
use std::fs;
use std::path::Path;

use crate::error::CouncilError;

fn load(prompts_dir: &Path, name: &str) -> Result<String, CouncilError> {
    let path = prompts_dir.join(name);
    fs::read_to_string(&path)
        .map(|s| s.trim().to_string())
        .map_err(|_| {
            CouncilError::FileNotFound(format!("Prompt file not found: {}", path.display()))
        })
}

pub fn discussion_prompt(
    prompts_dir: &Path,
    round_num: u32,
    total_rounds: u32,
) -> Result<String, CouncilError> {
    let round_file = format!("round-{}.md", round_num.min(3));
    let round_guidance = load(prompts_dir, &round_file)?;
    let engagement = load(prompts_dir, "engagement.md")?;
    let brevity = load(prompts_dir, "brevity.md")?;
    let response_format = load(prompts_dir, "response-format.md")?;

    Ok(format!(
        "You are participating in a council discussion, Round {} of {}.\n\n{}\n\n{}\n\n{}\n\n{}",
        round_num, total_rounds, round_guidance, engagement, brevity, response_format
    ))
}

pub fn vote_prompt(prompts_dir: &Path, question: &str) -> Result<String, CouncilError> {
    let template = load(prompts_dir, "vote.md")?;
    let engagement = load(prompts_dir, "engagement.md")?;
    let vote_format = load(prompts_dir, "vote-format.md")?;

    Ok(format!(
        "{}\n\n{}\n\n{}",
        template.replace("{question}", question),
        engagement,
        vote_format
    ))
}
